apps/labs/posts/free-threaded-python-rollout.md

```diff
@@ -0,0 +1,187 @@
+to work in CI and doing integration-level and real-world testing with users.
+
+
+## The road ahead & what our team will be working on
+
+Free-threaded CPython becoming the default, and eventually the only, build of
+CPython is several years away. What we're hoping to see, and help accomplish,
+is that for Python 3.13 many projects will work on compatibility and start
+releasing `cp313t` wheels on PyPI (and possibly nightly builds too, for projects
+with a lot of dependencies), so users and packages further downstream can start
+experimenting as well. After a full year of maturing support in the ecosystem
+and further improvements in performance in CPython itself, we should have a
+good picture of both the benefits and the remaining challenges with robustness.
+
+Our team (currently [Nathan](https://github.com/ngoldbaum),
+[Ken Jin](https://github.com/Fidget-Spinner),
+[Lysandros](https://github.com/lysnikolaou/),
+[Edgar](https://github.com/andfoy/), and
+[myself](https://github.com/rgommers/)) has now been working on this topic for
+a few months, starting at the bottom of the PyData stack (most effort so far
+has gone to NumPy, Cython, and CPython), and slowly working our way up from
+there.
+
+For each package, the approach has been similar so far - and a lot of that can
+be used as a template by others we think. The steps are roughly:
+
+1. Add a first CI job, usually Linux x86-64 with the latest Python 3.13
+   pre-release candidate, and ensure the test suite passes,
+2. Based on knowledge from maintainers, fix known issues with thread-safety and
+   shared/global state in native code,
+3. Add free-threaded support to the wheel build CI jobs, and start uploading
+   nightly wheels (if appropriate for the project),
+4. Do some stress testing locally and monitor CI jobs, and fix failures that
+   are observed (take the opportunity to add regression tests using `threading`
+   or `concurrent.futures.ThreadPoolExecutor`)
+5. Mark extension modules as supporting running without the GIL
+6. Move on to a next package (e.g., a key dependency) and using its test suite
+   to exercise the first package more, circling back to fix issues or address
+   follow-up actions as needed.
+
+Our main takeaway so far: it's challenging, but tractable! And fun as well:)
+
+We've only just scratched the surface, there'll be a lot to do - from key
+complex packages like PyO3 (of key importance for projects using Rust) and
+PyTorch, to the sheer volume of smaller packages with extension modules.
+The lessons we are learning, as far as they are reusable, are going into the
+documentation at
+[py-free-threading.github.io](https://py-free-threading.github.io).
+
+Furthermore, we'd like to spend time on whatever may be impactful in helping
+the ecosystem adopt free-threaded CPython, from answering questions
+to helping with debugging - please don't hesitate to reach out or ping us on
+GitHub!
```

Comment:
Maybe add a note that this repo is for ecosystem-wide coordination, so try to open project-specific issues in the project issue tracker.

Reply:
Ah good point - I didn't actually mention https://github.com/Quansight-Labs/free-threaded-compatibility/. Will extend this a bit.